File: src/main/java/com/baulsupp/oksocial/output/TestOutputHandler.java
package com.baulsupp.oksocial.output;

import com.google.common.collect.Lists;
import java.io.IOException;
import java.util.List;

public class TestOutputHandler<R> implements OutputHandler<R> {
  public final List<R> responses = Lists.newArrayList();
  public final List<Throwable> failures = Lists.newArrayList();
  public final List<String> stdout = Lists.newArrayList();

  @Override public void showOutput(R response) throws IOException {
    responses.add(response);
  }

  @Override public void showError(String s, Throwable e) {
    failures.add(e);
  }

  @Override public void info(String s) {
    stdout.add(s);
  }
}

File: src/main/kotlin/com/baulsupp/oksocial/output/TestOutputHandler.kt
package com.baulsupp.oksocial.output

class TestOutputHandler<R> : OutputHandler<R> {
  val responses: MutableList<R> = mutableListOf()
  val failures: MutableList<Throwable> = mutableListOf()
  val stdout: MutableList<String> = mutableListOf()

  override suspend fun showOutput(response: R) {
    responses.add(response)
  }

  override suspend fun showError(message: String?, e: Throwable?) {
    if (e != null) {
      failures.add(e)
    }
  }

  override fun info(message: String) {
    stdout.add(message)
  }
}
